### PR TITLE
Fix for btrfs package issue for Ubuntu distribution

### PR DESCRIPTION
**Pull request by Naresh-ibm**
The package for btrfs file system is different in ubuntu. hence it
was failing while running for btrfs file system. added the code to
fix the same issue. now it is working fine.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[bonnie_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/964309/bonnie_log.txt)
[fiotest_job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/964307/fiotest_job.txt)
[fs_mark_job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/964308/fs_mark_job.txt)
[tiobench_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/964310/tiobench_log.txt)

File: io/disk/bonnie.py [hunk 1]
```diff
@@ -47,6 +48,12 @@ def setUp(self):
          http://www.coker.com.au/bonnie++/experimental/bonnie++-1.03e.tgz
         """
 
+        smm = SoftwareManager()
+        if distro.detect().name == 'Ubuntu':
+            if not smm.check_installed("btrfs-tools") and not \
+                    smm.install("btrfs-tools"):
+                self.error('btrfs-tools is needed for the test to be run')
```
Comment (narasimhan-v) [inline io/disk/bonnie.py]: `self.skip()`

File: io/disk/bonnie.py [hunk 2]
```diff
@@ -47,6 +48,12 @@ def setUp(self):
          http://www.coker.com.au/bonnie++/experimental/bonnie++-1.03e.tgz
         """
 
+        smm = SoftwareManager()
+        if distro.detect().name == 'Ubuntu':
```
Comment (narasimhan-v) [inline io/disk/bonnie.py]: Is the package name same for all versions of ubuntu? Can you check plese

Comment (Naresh-ibm) [inline io/disk/bonnie.py]: I have verified in ubuntu17.04 and it is same.

**Comment by Naresh-ibm** [comment]
Amended the changes suggested.

File: io/disk/tiobench.py [hunk 3]
```diff
@@ -44,9 +44,13 @@ def setUp(self):
...
-        s_mngr = SoftwareManager()
-        if not s_mngr.check_installed("gcc") and not s_mngr.install("gcc"):
-            self.error('Gcc is needed for the test to be run')
+        smm = SoftwareManager()
+        if distro.detect().name == 'Ubuntu':
+            if not smm.check_installed("btrfs-tools") and not \
+                    smm.install("btrfs-tools"):
+                self.skip('btrfs-tools is needed for the test to be run')
+        if not smm.check_installed("gcc") and not smm.install("gcc"):
+            self.skip('Gcc is needed for the test to be run')
```
Comment (PraveenPenguin) [inline io/disk/tiobench.py]: can this be like 

```
packages=[gcc]
if distro.detect().name == 'Ubuntu':
    packages.extend(['btrfs-tools']
        for package in packages:
            if not sm.check_installed(package) and not sm.install(package):
                self.skip("Fail to install %s required for this test." %
                           package)
```

File: io/disk/tiobench.py [hunk 4]
```diff
@@ -44,9 +44,13 @@ def setUp(self):
...
         locations = ["https://github.com/mkuoppal/tiobench/archive/master.zip"]
         tarball = self.fetch_asset("tiobench.zip", locations=locations)
```
Comment (PraveenPenguin) [inline io/disk/tiobench.py]: please use expire param in` fetch_asset` it help to download only delta

File: io/disk/tiobench.py (hunk 3 above)
Comment (narasimhan-v) [inline io/disk/tiobench.py]: btrfs package should be installed only if `fstype` is btrfs.
Please refer: https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/io/disk/lvsetup.py

**Comment by narasimhan-v** [comment]
@Naresh-ibm This PR has many changes which should be addressed by separate commits.
Please split them.

**Comment by PraveenPenguin** [comment]
@narasimhan-v yup make sense

**Comment by Naresh-ibm** [comment]
closing this in favour of #465

**Pull request closed without merging**